pgml-extension/src/bindings/transformers/transformers.py

```diff
@@ -956,3 +970,610 @@ def generate(model_id, data, config):
             )
             all_preds.extend(decoded_preds)
     return all_preds
+
+
+#######################
+# LLM Fine-Tuning
+#######################
+
+
+class PGMLCallback(TrainerCallback):
+    "A callback that prints a message at the beginning of training"
+
+    def __init__(self, project_id, model_id):
+        self.project_id = project_id
+        self.model_id = model_id
+
+    def on_log(self, args, state, control, logs=None, **kwargs):
+        _ = logs.pop("total_flos", None)
+        if state.is_local_process_zero:
+            logs["step"] = state.global_step
+            logs["max_steps"] = state.max_steps
+            logs["timestamp"] = str(datetime.now())
+            print_info(json.dumps(logs))
```

Comment:
If you use use `print()`, this will appear in Postgres logs. It won't be pretty, but we can add a function that formats it correctly.

Reply:
I will add indent in json.dumps() to pretty print.